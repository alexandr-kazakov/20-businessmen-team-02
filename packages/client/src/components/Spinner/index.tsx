import React from 'react'

import styles from './styles.module.scss'

export const Spinner: React.FC = () => <div className={styles.spinner} />
