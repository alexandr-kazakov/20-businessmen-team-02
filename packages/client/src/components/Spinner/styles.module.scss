@keyframes spin {
  from {
    transform: rotate(0turn);
  }

  to {
    transform: rotate(1turn);
  }
}

.spinner {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;

  &::after {
    content: '';
    width: 50px;
    height: 50px;
    border: 5px solid #ddd;
    border-top-color: #d9534f;
    border-radius: 50%;
    animation: spin 0.75s ease infinite;
  }
}
