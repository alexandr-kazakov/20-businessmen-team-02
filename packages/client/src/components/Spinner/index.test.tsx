import React from 'react'
import { render } from '@testing-library/react'
import { Spinner } from '@/components/Spinner/index'

test('should render Spinner component', () => {
  const {asFragment} = render(<Spinner/>)

  expect(asFragment()).toMatchSnapshot()
})