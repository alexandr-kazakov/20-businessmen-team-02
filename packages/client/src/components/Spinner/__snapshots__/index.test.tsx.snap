// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Spinner component 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
